### ISSUE TITLE
Allow registering custom widgets

### ISSUE THREAD

**Issue by n1k0** [mozilla-services/react-jsonschema-form#63]
Self-explanatory dream API:

```js
import Form from "react-jsonschema-form";

const MyComponent = (props) => {
  // ....
};

const schema = {
  type: "string"
};

const widgets = {
  myComponent: MyComponent
};

const uiSchema = {
  "ui:widget": "myComponent"
};

render((<Form 
  schema={schema} 
  uiSchema={uiSchema} 
  widgets={widgets}
/>), document.querySelector("main"))
```

Context: as `uiSchema` should be defined using JSON (and no necessarily JavaScript), we need to expose a way to register a custom component and associate it with a string unique identifier we can later reference from the JSON definition.
 
/cc @leplatrem

**Comment by n1k0** [comment]
Basic PoC in #67

**Comment by n1k0** [comment]
Fixed by #67.
